implementations/rust/ockam/ockam_app/src/invitations/commands.rs

```diff
@@ -305,23 +312,83 @@ async fn create_inlet(inlet_data: &InletDataFromInvitation) -> crate::Result<Soc
     info!(
         from = from_str,
         to = service_route,
-        "Created tcp-inlet for accepted invitation"
+        "Created TCP inlet for accepted invitation"
     );
     Ok(from)
 }
 
+pub(crate) async fn disconnect_tcp_inlet<R: Runtime>(
+    app: AppHandle<R>,
+    invitation_id: &str,
+) -> crate::Result<()> {
+    let invitation_state: State<'_, SyncInvitationsState> = app.state();
+    let mut writer = invitation_state.write().await;
+    if let Some(inlet) = writer.accepted.inlets.get_mut(invitation_id) {
+        if !inlet.enabled {
+            debug!(node = %inlet.node_name, alias = %inlet.alias, "TCP inlet was already disconnected");
+            return Ok(());
+        }
+        inlet.disable();
+        let local_node_name = &inlet.node_name;
+        let alias = &inlet.alias;
+        debug!(node = %local_node_name, %alias, "Deleting TCP inlet");
+        let _ = duct::cmd!(
+            &cli_bin()?,
+            "--no-input",
+            "tcp-inlet",
+            "delete",
+            alias,
+            "--at",
+            local_node_name,
+            "--yes"
+        )
+        .stderr_null()
+        .stdout_capture()
+        .run()
+        .map_err(
+            |e| warn!(%e, node = %local_node_name, alias = %alias, "Failed to delete TCP inlet"),
+        );
+        info!(
+            node = %local_node_name, %alias,
+            "Disconnected TCP inlet for accepted invitation"
+        );
+    }
+    Ok(())
+}
+
+pub(crate) async fn enable_tcp_inlet<R: Runtime>(
+    app: AppHandle<R>,
+    invitation_id: &str,
+) -> crate::Result<()> {
+    let invitation_state: State<'_, SyncInvitationsState> = app.state();
+    let mut writer = invitation_state.write().await;
+    if let Some(inlet) = writer.accepted.inlets.get_mut(invitation_id) {
+        if inlet.enabled {
+            debug!(node = %inlet.node_name, alias = %inlet.alias, "TCP inlet was already enabled");
+            return Ok(());
+        }
+        inlet.enable();
+        app.trigger_global(super::events::REFRESH_INVITATIONS, None);
+        info!(node = %inlet.node_name, alias = %inlet.alias, "Enabled TCP inlet");
+    }
+    Ok(())
+}
+
 #[derive(Debug)]
-struct InletDataFromInvitation {
+pub(crate) struct InletDataFromInvitation {
+    pub enabled: bool,
     pub local_node_name: String,
     pub service_name: String,
     pub service_route: String,
     pub enrollment_ticket_hex: Option<String>,
+    pub socket_addr: Option<SocketAddr>,
 }
 
 impl InletDataFromInvitation {
     pub fn new(
         cli_state: &CliState,
         invitation: &InvitationWithAccess,
+        inlets: &HashMap<String, Inlet>,
```

Comment:
This constructor feels wrong to me. That function should probably somewhere else, and, as a result, return a new `ServiceInlet`, ready to be added to the list of services.